### PR TITLE
update automatic sample name generation mechanism

### PR DESCRIPTION
**Pull request by cyrielo**
# Description
This PR generates a sample name from the file path and gives the user ability to edit the name

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Impacted Areas in Application
List general components of the application that this PR will affect:

Which pages?

- [x] Single Upload Page

Specific components 

* SampleUpload

# Testing Script:

###Single Upload
* Using an AWS path that doesn't exist throws an error
* Successfully uploaded files from a folder (use test host)
* Uploading a duplicate file throws an error
* Pipeline gets initiated successfully 

# Checklist:

- [x] I have run the testing script to make sure current functionality is unchanged
- [x] I have done relevant tests that prove my fix is effective or that my feature works
- [x] I have spent time testing out edge cases for my feature
- [x] I have updated the test script or pull request template if necessary
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/assets/javascripts/components/sampleUpload.jsx [hunk 1]
```diff
@@ -778,6 +803,32 @@ class SampleUpload extends React.Component {
...
+                      <div className='field-title'>
+                        <div className='read-count-label'>
+                          Sample name
+                        </div>
+                      </div>
+                    </div>
+                  </div>
+                  <div className='row input-row'>
+                    <div className='col no-padding s12'>
+                      <input type='text' ref='sample_name' className='browser-default' value={this.state.generatedSampleName} onChange={ (e) => {
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/sampleUpload.jsx]: I think sample name should be above the admin options.

File: app/assets/javascripts/components/sampleUpload.jsx [hunk 2]
```diff
@@ -778,6 +803,32 @@ class SampleUpload extends React.Component {
...
+                        this.setState({
+                            generatedSampleName: e.target.value.trim()
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/sampleUpload.jsx]: I think you should also set a state so if people change Read path, the sample name  entered by the user won't be  overwritten. Also I think the function should be extracted out.

File: app/assets/javascripts/components/sampleUpload.jsx [hunk 3]
```diff
@@ -57,7 +58,8 @@ class SampleUpload extends React.Component {
       selectedMemory: this.selected.memory || '',
       id: this.selected.id,
       errors: {},
-      adminGenomes
+      adminGenomes,
+      generatedSampleName: ''
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/sampleUpload.jsx]: this should just be called sample name.

File: app/assets/javascripts/components/sampleUpload.jsx [hunk 4]
```diff
@@ -412,6 +425,17 @@ class SampleUpload extends React.Component {
     }
   }
 
+  updateGeneratedSampleName(e) {
+    let value = e.target.value.trim();
```
Comment (yunfangjuan) [inline app/assets/javascripts/components/sampleUpload.jsx]: as mentioned, only do this when a user not editing the name.

File: app/assets/javascripts/components/sampleUpload.jsx (hunk 1 above)
Comment (cyrielo) [reply app/assets/javascripts/components/sampleUpload.jsx]: Yeah we didn't want to have the user manually type in a sample name, so we put it at the bottom so we can generate it and have the user edit it

Comment (yunfangjuan) [reply app/assets/javascripts/components/sampleUpload.jsx]: but it's below the admin options. users should be able to edit it. 
it's odd to have <user options> <admin options> and then <user options> again.

**Comment by yunfangjuan** [comment]
Can you please also move the sample name field to be above the admin options?

**Pull request merged**